File: dmn-signavio/src/test/resources/dmn2java/exported/complex/expected/dmn/example-credit-decision/ProcessPriorIssuesRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "processPriorIssues"})
public class ProcessPriorIssuesRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private java.math.BigDecimal processPriorIssues;

    public ProcessPriorIssuesRuleOutput(boolean matched) {
        super(matched);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("processPriorIssues")
    public java.math.BigDecimal getProcessPriorIssues() {
        return this.processPriorIssues;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("processPriorIssues")
    public void setProcessPriorIssues(java.math.BigDecimal processPriorIssues) {
        this.processPriorIssues = processPriorIssues;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ProcessPriorIssuesRuleOutput other = (ProcessPriorIssuesRuleOutput) o;
        if (this.getProcessPriorIssues() != null ? !this.getProcessPriorIssues().equals(other.getProcessPriorIssues()) : other.getProcessPriorIssues() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getProcessPriorIssues() != null ? this.getProcessPriorIssues().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", processPriorIssues='%s'", processPriorIssues));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-test-cases/signavio/dmn/dmn2java/expected/proto/proto3/kotlin/dmn/example-credit-decision/ProcessPriorIssuesRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "processPriorIssues"])
class ProcessPriorIssuesRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("processPriorIssues")
    var processPriorIssues: java.math.BigDecimal? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as ProcessPriorIssuesRuleOutput
        if (if (this.processPriorIssues != null) this.processPriorIssues != other.processPriorIssues else other.processPriorIssues != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.processPriorIssues != null) this.processPriorIssues.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", processPriorIssues='%s'", processPriorIssues))
        result_.append(")")
        return result_.toString()
    }
}
